llmfoundry/models/hf/hf_causal_lm.py

```diff
@@ -92,6 +89,9 @@ def __init__(self, om_model_config: Union[DictConfig,
                     'which is not significantly slower and not compatible with the LLM foundry training code, rather than the code release by MosaicML.'
                 )
 
+            if not om_model_config.get('use_train_metrics', True):
+                train_metrics = []
+
```

Comment:
```suggestion
            use_train_metrics = om_model_config.get('use_train_metrics', True)
        else:
            use_train_metrics = True

        train_metrics = [LanguageCrossEntropy(),
                         LanguagePerplexity()] if use_train_metrics else []
```

Reply:
The spacing here doesn't quite work since `train_metrics` needs to be scoped outside the `if` (which is a very long `if`). This is due to the cases where the config is not a DictConfig (the typing and codepath here needs to be cleaned up...). So I will resolve